File: evals/dispatch/fixtures/ufs_ok.go
// Pins the Go lane carve-outs in audits/ufs.sh — remove either and this
// fixture starts failing.
//
//   `const ( ... )`  the binding keyword is stated ONCE, on the opening line,
//                    so every member below it is a definition carrying no
//                    `const` of its own. Two names sharing one value across
//                    two domains is the case the single-line carve-out allows.
//   struct tags      backtick-quoted halves are read by reflection; a const
//                    cannot appear inside one, so no rename can fix a repeat.
package fixture

const (
	runnerActive = "active"
	leaseActive  = "active"
)

type Row struct {
	Name string `json:"repeated_tag_name" yaml:"repeated_tag_name"`
}

func Active(lease bool) string {
	if lease {
		return leaseActive
	}
	return runnerActive
}

File: evals/dispatch/fixtures/ufs_ok.rs
// Every literal here is either bound to a name or unfixable by binding one.
// Pins the Rust lane carve-outs in audits/ufs.sh — remove any one of them and
// this fixture starts failing, which is the point of it.
//
//   `static` bindings   two names, one value, two domains — the same case the
//                       single-line const carve-out already allows
//   `#[...]` attributes a literal token the language requires; no const is
//                       accepted in that position, so no rename can fix it
//   `#[cfg(test)]`      Rust keeps unit tests inside the file they cover, so
//                       fixture keys would otherwise read as production debt
static WIRE_PING: &str = "PING";
static HEALTH_PROBE: &str = "PING";

#[derive(Debug)]
pub struct Probe {
    #[serde(rename = "repeated_attribute_literal")]
    pub sent: String,
    #[serde(rename = "repeated_attribute_literal")]
    pub seen: String,
}

pub fn probe(reply: &str) -> bool {
    reply == WIRE_PING || reply == HEALTH_PROBE
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn repetition_here_is_fixture_data() {
        assert!(probe("repeated_fixture_key"));
        assert!(probe("repeated_fixture_key"));
    }
}
